fiber/fiber.c

```diff
@@ -119,13 +119,24 @@ int fiber_wait_all()
     return FIBER_NOERROR;
 }
 
+/* A simple solution to avoid unexpected output by the buffered I/O printf is
```

Comment:
You shall explain why this wrapper is "safe."

Reply:
Actually, I am thinking to change the name for the macro.  I use the word "safe" to represent that the output can show as we may want. But as far as I know, `printf` is said to be thread-safe in the Linux manual. That means it won't corrupt anything when using in the multithreading. However, some interleaving that doesn't change the outcome is still allowed to happen, which can produce the mixing output between threads. 

So I am not sure if the name `print_safe` will mislead someone to the concept that `printf` is not thread-safe. Do I misunderstand something or any advice for this?

Reply:
It was meant to be "safe." Instead, its intention is to eliminate the impact of buffered I/O, which causes the unexpected sequence of strings.

Reply:
While the `printf` provided by glibc is a buffered output function, the proposed change does "direct" format string manipulation. Therefore, you can rename `printf_safe` to `printf_unbuffered` and then use `#define printf printf_unbuffered` for build-time substitution.